mysite/psa/tests.py

```diff
@@ -816,3 +818,68 @@ def test_optional_expiry_interval_param(self):
         _wrapped_view = preview_access(decorated_view, expiry_interval=interval)
         _wrapped_view(self.request)
         self.request.session.set_expiry.assert_called_once_with(interval)
+
+
+class ForgotPasswordTest(TestCase):
+    """
+    Functional test reset password
+    """
+
+    def setUp(self):
+        self.user = User(username='user')
+        self.user.set_password('user')
+        self.user.save()
+        self.token = TokenForgotPassword()
+        self.token.user = self.user
+        self.token.next_url = '/'
+        self.token.save()
+
+    def test_reset_password(self):
+        test_password = 'new_pass'
+        post_data = {
+            'password': test_password,
+            'password1': test_password
+        }
+        old_pass = User.objects.get(id=self.user.id).password
```

Comment:
Hmm, I think you can use two times less code than it is now, lets discuss, something like this:

    old_pass = User.objects.get(id=self.user.id).password
    test_url = '/reset-pass/{0}/'.format(self.token.token)
    res = self.client.post(test_url, post_data)
    self.assertIsInstance(res, HttpResponse)
    self.assertEqual(res.status_code, 200)
    self.assertTrue('Your password has been changed' in res.content)
    self.assertIn('next_url', res.context)
    new_pass = User.objects.get(id=self.user.id).password
    self.assertNotEquals(old_pass, new_pass)